shared/src/main/kotlin/com/egm/stellio/shared/model/ExpandedEntity.kt

```diff
@@ -110,22 +114,41 @@ class ExpandedEntity private constructor(
         }
     }
 
-    fun checkPropertiesHaveAtMostOneDefaultInstance() {
-        properties.forEach { property ->
-            if (property.value.count { !it.containsKey(NGSILD_DATASET_ID_PROPERTY) } > 1)
-                throw BadRequestDataException("Property ${property.key.extractShortTypeFromExpanded()} can't have more than one default instance")
+    fun checkAttributesHaveAtMostOneDefaultInstance() {
+        checkAttributesHaveAtMostOneDefaultInstance(NGSILD_PROPERTY_TYPE)
```

Comment:
Best would be to go with something like : 
```kotlin
fun checkAttributesHaveAtMostOneDefaultInstance() {
	properties.checkAttributesHaveAtMostOneDefaultInstance()?.let { throw BadRequestDataException("Property ${it.extractShortTypeFromExpanded()} can't have more than one default instance") }
	relationships.checkAttributesHaveAtMostOneDefaultInstance()?.let { throw BadRequestDataException("Relationship ${it.extractShortTypeFromExpanded()} can't have more than one default instance") }
}

private fun Map<String, List<Map<String, List<Any>>>>.checkAttributesHaveAtMostOneDefaultInstance(): String? {
    this.forEach { attribute ->
        if (attribute.value.count { !it.containsKey(NGSILD_DATASET_ID_PROPERTY) } > 1) {
        	return attribute.key
        }
    }
}
```
In my opinion it's better as the shared method stays neutral whether the type is.

Reply:
And I really like the Kotlin extensions